src/main/java/org/jenkinsci/plugins/nomad/NomadWorkerTemplate.java

```diff
@@ -1,155 +1,81 @@
 package org.jenkinsci.plugins.nomad;
 
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+
 import hudson.Extension;
 import hudson.Util;
 import hudson.model.Describable;
 import hudson.model.Descriptor;
-import hudson.model.Label;
 import hudson.model.Node;
 import hudson.model.labels.LabelAtom;
 import hudson.util.Secret;
 import jenkins.model.Jenkins;
-import org.apache.commons.lang.StringUtils;
-import org.kohsuke.stapler.DataBoundConstructor;
-
-import java.util.Collections;
-import java.util.List;
-import java.util.Set;
-import java.util.logging.Logger;
 
 public class NomadWorkerTemplate implements Describable<NomadWorkerTemplate> {
 
     private static final String SLAVE_PREFIX = "jenkins";
-    private static final Logger LOGGER = Logger.getLogger(NomadWorkerTemplate.class.getName());
 
+    // persistent fields
+    private final String prefix;
     private final int idleTerminationInMinutes;
-    private final Boolean reusable;
+    private final boolean reusable;
     private final int numExecutors;
-
-    private final String prefix;
-    private final int cpu;
-    private final int memory;
-    private final int disk;
-    private final int priority;
     private final String labels;
-    private final List<? extends NomadConstraintTemplate> constraints;
-    private final String region;
-    private final String remoteFs;
-    private final Boolean useRawExec;
-    private final String image;
-    private final Boolean privileged;
-    private final String network;
-    private final String username;
-    private final Secret password;
-    private final String prefixCmd;
-    private final Boolean forcePull;
-    private final String hostVolumes;
-    private final String switchUser;
-    private final Node.Mode mode;
-    private final List<? extends NomadPortTemplate> ports;
-    private final String extraHosts;
-    private final String dnsServers;
-    private final String securityOpt;
-    private final String capAdd;
-    private final String capDrop;
-    private final String datacenters;
-    private final String vaultPolicies;
-    private final Set<LabelAtom> labelSet;
-    private final List<? extends NomadDevicePluginTemplate> devicePlugins;
-    private String driver;
+    private final String jobTemplate;
+
+    // legacy fields (we have to keep them for backward compatibility)
```

Comment:
I guess they are all deprecated now? Can they all be marked as such if it's the case?

Reply:
What you have in mind? I could replace legacy by deprecated...

Reply:
re:

> What is about the SpotBugs findings (UUF_UNUSED_FIELD)? I don't know how this could be solved? Is there a way to ignore them?

Adding `@Deprecated` actually remove the SpotBugs warning ;)